justfile

```diff
@@ -15,11 +15,25 @@ start:
     #!/usr/bin/env bash
     set -euxo pipefail
 
-    if ! minikube delete; then
-        echo "Detected a fresh minikube environment."
-    fi
-
-    minikube start --mount --mount-string="$PWD:/mnt/src"
+    # Function to check if minikube is running
+    check_minikube() {
+        minikube status | grep -q "Running" && echo "Minikube is already running" || minikube start --mount --mount-string="$PWD:/mnt/src"
```

Comment:
This check  does not catch kubectl issues and so it leaves warnet in an unusable/intermediate state after certain kinds of errors.

For example, `just installlogging` fails. It also fails for me on main, but on main the failure looks like this:

``` shell
Error: failed post-install: 1 error occurred:
        * timed out waiting for the condition
```

<details>
<summary>On remove-compose it looks like this</summary>

```` shell
(.venv) grant@remove-compose:~/warnet$ kubectl get all
NAME                     READY   STATUS    RESTARTS   AGE
pod/rpc-0                1/1     Running   0          62m
pod/warnet-tank-000000   1/1     Running   0          61m
pod/warnet-tank-000001   1/1     Running   0          61m
pod/warnet-tank-000002   1/1     Running   0          61m
pod/warnet-tank-000003   1/1     Running   0          61m
pod/warnet-tank-000004   1/1     Running   0          61m
pod/warnet-tank-000005   1/1     Running   0          61m
pod/warnet-tank-000006   1/1     Running   0          61m
pod/warnet-tank-000007   1/1     Running   0          61m
pod/warnet-tank-000008   1/1     Running   0          61m
pod/warnet-tank-000009   1/1     Running   0          61m
pod/warnet-tank-000010   1/1     Running   0          61m
pod/warnet-tank-000011   1/1     Running   0          61m

NAME                                 TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                            AGE
service/rpc                          ClusterIP   None             <none>        9276/TCP                                           62m
service/warnet-tank-000000-service   ClusterIP   10.97.130.76     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m
service/warnet-tank-000001-service   ClusterIP   10.98.227.82     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m
service/warnet-tank-000002-service   ClusterIP   10.104.158.18    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m
service/warnet-tank-000003-service   ClusterIP   10.102.166.85    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m
service/warnet-tank-000004-service   ClusterIP   10.103.177.164   <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m
service/warnet-tank-000005-service   ClusterIP   10.101.119.229   <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m
service/warnet-tank-000006-service   ClusterIP   10.109.216.81    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m
service/warnet-tank-000007-service   ClusterIP   10.100.56.101    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m
service/warnet-tank-000008-service   ClusterIP   10.101.155.49    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m
service/warnet-tank-000009-service   ClusterIP   10.106.24.140    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m
service/warnet-tank-000010-service   ClusterIP   10.96.21.131     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m
service/warnet-tank-000011-service   ClusterIP   10.99.47.99      <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   61m

NAME                   READY   AGE
statefulset.apps/rpc   1/1     62m
(.venv) grant@remove-compose:~/warnet$ just installlogging
./src/templates/k8s/install_logging.sh
./src/templates/k8s/install_logging.sh: line 6: helm: command not found
error: Recipe `installlogging` failed on line 106 with exit code 127
(.venv) grant@remove-compose:~/warnet$ cd ..
(.venv) grant@remove-compose:~$ $ curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3
-bash: $: command not found
(.venv) grant@remove-compose:~$  curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3
(.venv) grant@remove-compose:~$ chmod 700 get_helm.sh
(.venv) grant@remove-compose:~$  ./get_helm.sh
Downloading https://get.helm.sh/helm-v3.15.2-linux-amd64.tar.gz
Verifying checksum... Done.
Preparing to install helm into /usr/local/bin
helm installed into /usr/local/bin/helm
(.venv) grant@remove-compose:~$ cd warnet/
(.venv) grant@remove-compose:~/warnet$ just installlogging
./src/templates/k8s/install_logging.sh
"grafana" has been added to your repositories
"prometheus-community" has been added to your repositories
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "grafana" chart repository
...Successfully got an update from the "prometheus-community" chart repository
Update Complete. ⎈Happy Helming!⎈
Release "loki" does not exist. Installing it now.
W0614 17:04:49.049821  182132 reflector.go:470] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: watch of *unstructured.Unstructured ended with: an error on the server ("unable to decode an event from the watch stream: http2: client connection lost") has prevented the request from succeeding
W0614 17:04:59.912818  182132 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
I0614 17:04:59.912995  182132 trace.go:236] Trace[236677641]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 17:04:49.905) (total time: 10007ms):
Trace[236677641]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout 10007ms (17:04:59.912)
Trace[236677641]: [10.007180961s] [10.007180961s] END
E0614 17:04:59.923008  182132 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
W0614 17:05:13.052188  182132 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
I0614 17:05:13.052338  182132 trace.go:236] Trace[801429994]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 17:05:03.051) (total time: 10000ms):
Trace[801429994]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout 10000ms (17:05:13.052)
Trace[801429994]: [10.000945787s] [10.000945787s] END
E0614 17:05:13.052368  182132 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
W0614 17:05:27.977998  182132 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
I0614 17:05:27.978096  182132 trace.go:236] Trace[440531367]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 17:05:17.972) (total time: 10005ms):
Trace[440531367]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout 10005ms (17:05:27.977)
Trace[440531367]: [10.005699513s] [10.005699513s] END
E0614 17:05:27.978122  182132 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
W0614 17:05:47.620730  182132 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
I0614 17:05:47.620820  182132 trace.go:236] Trace[2140840504]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 17:05:37.617) (total time: 10003ms):
Trace[2140840504]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout 10003ms (17:05:47.620)
Trace[2140840504]: [10.003453987s] [10.003453987s] END
E0614 17:05:47.620848  182132 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
W0614 17:06:21.392621  182132 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
I0614 17:06:21.392728  182132 trace.go:236] Trace[1432321337]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 17:06:11.390) (total time: 10002ms):
Trace[1432321337]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout 10002ms (17:06:21.392)
Trace[1432321337]: [10.002149653s] [10.002149653s] END
E0614 17:06:21.392758  182132 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
W0614 17:07:21.645121  182132 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
I0614 17:07:21.645213  182132 trace.go:236] Trace[916629026]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 17:07:11.636) (total time: 10008ms):
Trace[916629026]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout 10008ms (17:07:21.645)
Trace[916629026]: [10.008849836s] [10.008849836s] END
E0614 17:07:21.645239  182132 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
W0614 17:08:07.036535  182132 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
I0614 17:08:07.036631  182132 trace.go:236] Trace[1339682061]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 17:07:57.032) (total time: 10004ms):
Trace[1339682061]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout 10004ms (17:08:07.036)
Trace[1339682061]: [10.004209039s] [10.004209039s] END
E0614 17:08:07.036670  182132 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
W0614 17:09:02.793774  182132 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
I0614 17:09:02.794141  182132 trace.go:236] Trace[16137022]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 17:08:52.792) (total time: 10001ms):
Trace[16137022]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout 10001ms (17:09:02.793)
Trace[16137022]: [10.001257394s] [10.001257394s] END
E0614 17:09:02.794261  182132 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=4258": net/http: TLS handshake timeout
Error: failed post-install: 1 error occurred:
        * timed out waiting for the condition


error: Recipe `installlogging` failed on line 106 with exit code 1
````

</details>

<details>

<summary>After which , running `just start` again does not work because the  Tanks do not come up.</summary>

```` shell
  (.venv) grant@remove-compose:~/warnet$ just start
  + check_minikube
  + grep -q Running
  + minikube status
  + minikube start --mount --mount-string=/home/grant/warnet:/mnt/src
  😄  minikube v1.33.1 on Debian 12.5 (amd64)
  ✨  Using the docker driver based on existing profile
  👍  Starting "minikube" primary control-plane node in "minikube" cluster
  🚜  Pulling base image v0.0.44 ...
  🏃  Updating the running docker "minikube" container ...
  🐳  Preparing Kubernetes v1.30.0 on Docker 26.1.1 ...
  🔎  Verifying Kubernetes components...
      ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
  🌟  Enabled addons: storage-provisioner, default-storageclass
  🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
  + docker build -t warnet/dev -f src/templates/rpc/Dockerfile_rpc_dev . --load
  [+] Building 1.0s (11/11) FINISHED                                                                                                     docker:default
   => [internal] load build definition from Dockerfile_rpc_dev                                                                                     0.0s
   => => transferring dockerfile: 1.04kB                                                                                                           0.0s
   => [internal] load metadata for docker.io/library/python:3.12-slim                                                                              0.7s
   => [internal] load .dockerignore                                                                                                                0.0s
   => => transferring context: 173B                                                                                                                0.0s
   => [1/6] FROM docker.io/library/python:3.12-slim@sha256:2fba8e70a87bcc9f6edd20dda0a1d4adb32046d2acbca7361bc61da5a106a914                        0.0s
   => [internal] load build context                                                                                                                0.0s
   => => transferring context: 146B                                                                                                                0.0s
   => CACHED [2/6] RUN apt-get update &&     apt-get install -y procps openssh-client &&     rm -rf /var/lib/apt/lists/*                           0.0s
   => CACHED [3/6] RUN pip install uv                                                                                                              0.0s
   => CACHED [4/6] WORKDIR /root/warnet                                                                                                            0.0s
   => CACHED [5/6] COPY src/templates/rpc/entrypoint.sh /usr/local/bin/                                                                            0.0s
   => CACHED [6/6] RUN chmod +x /usr/local/bin/entrypoint.sh                                                                                       0.0s
   => exporting to image                                                                                                                           0.0s
   => => exporting layers                                                                                                                          0.0s
   => => writing image sha256:de81fe65284ccd15a761ecac835c88bc3335ad408fea8e6afaeb543ac15bc482                                                     0.0s
   => => naming to docker.io/warnet/dev                                                                                                            0.0s
  + minikube image load warnet/dev
  + kubectl apply -f src/templates/rpc/namespace.yaml
  namespace/warnet unchanged
  namespace/warnet-logging unchanged
  + kubectl apply -f src/templates/rpc/rbac-config.yaml
  role.rbac.authorization.k8s.io/pod-reader-exec unchanged
  rolebinding.rbac.authorization.k8s.io/read-exec-pods unchanged
  clusterrole.rbac.authorization.k8s.io/crd-viewer unchanged
  clusterrolebinding.rbac.authorization.k8s.io/crd-viewer-binding unchanged
  + kubectl apply -f src/templates/rpc/warnet-rpc-service.yaml
  service/rpc configured
  + kubectl apply -f src/templates/rpc/warnet-rpc-statefulset-dev.yaml
  statefulset.apps/rpc unchanged
  + kubectl config set-context --current --namespace=warnet
  Context "minikube" modified.
  + check_warnet_rpc
  + kubectl get pods --all-namespaces
  + grep -q bitcoindevproject/warnet-rpc
  + kubectl get pod rpc-0 --namespace=warnet
  NAME    READY   STATUS    RESTARTS     AGE
  rpc-0   0/1     Running   3 (7s ago)   83m
  + echo '⏲️ This could take a minute or so.'
  ⏲️ This could take a minute or so.
  + kubectl wait --for=condition=Ready --timeout=2m pod rpc-0
  pod/rpc-0 condition met
  + echo Done...
  Done...
  (.venv) grant@remove-compose:~/warnet$ kubectl get all
  NAME                     READY   STATUS      RESTARTS      AGE
  pod/rpc-0                1/1     Running     3 (90s ago)   85m
  pod/warnet-tank-000000   0/1     Completed   0             84m
  pod/warnet-tank-000001   0/1     Completed   0             84m
  pod/warnet-tank-000002   0/1     Completed   0             84m
  pod/warnet-tank-000003   0/1     Completed   0             84m
  pod/warnet-tank-000004   0/1     Completed   1             84m
  pod/warnet-tank-000005   0/1     Completed   1             84m
  pod/warnet-tank-000006   0/1     Completed   1             84m
  pod/warnet-tank-000007   0/1     Completed   0             84m
  pod/warnet-tank-000008   0/1     Completed   0             84m
  pod/warnet-tank-000009   0/1     Completed   0             84m
  pod/warnet-tank-000010   0/1     Completed   0             84m
  pod/warnet-tank-000011   0/1     Completed   0             84m

  NAME                                 TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                            AGE
  service/rpc                          ClusterIP   None             <none>        9276/TCP                                           85m
  service/warnet-tank-000000-service   ClusterIP   10.97.130.76     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m
  service/warnet-tank-000001-service   ClusterIP   10.98.227.82     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m
  service/warnet-tank-000002-service   ClusterIP   10.104.158.18    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m
  service/warnet-tank-000003-service   ClusterIP   10.102.166.85    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m
  service/warnet-tank-000004-service   ClusterIP   10.103.177.164   <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m
  service/warnet-tank-000005-service   ClusterIP   10.101.119.229   <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m
  service/warnet-tank-000006-service   ClusterIP   10.109.216.81    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m
  service/warnet-tank-000007-service   ClusterIP   10.100.56.101    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m
  service/warnet-tank-000008-service   ClusterIP   10.101.155.49    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m
  service/warnet-tank-000009-service   ClusterIP   10.106.24.140    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m
  service/warnet-tank-000010-service   ClusterIP   10.96.21.131     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m
  service/warnet-tank-000011-service   ClusterIP   10.99.47.99      <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   84m

  NAME                   READY   AGE
  statefulset.apps/rpc   1/1     85m

````

</details>

<details>

<summary>And, `network start`  does not bring them up.</summary>


```` shell
(.venv) grant@remove-compose:~/warnet$ warcli network start --force
Handling connection for 9276
Warnet:
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Temp dir                    ┃ Bitcoin network ┃ Docker network ┃ Subnet      ┃ Graph                                   ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ /root/.warnet/warnet/warnet │ regtest         │ warnet         │ 100.0.0.0/8 │ MultiDiGraph with 12 nodes and 12 edges │
└─────────────────────────────┴─────────────────┴────────────────┴─────────────┴─────────────────────────────────────────┘

Tanks:
┏━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━┓
┃ Index ┃ Version                        ┃ IPv4            ┃ bitcoin conf                 ┃ tc_netem ┃ LN Image ┃
┡━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━┩
│ 0     │ 27.0                           │ 100.16.129.14   │ -uacomment=w0                │ None     │          │
│ 1     │ 27.0                           │ 100.52.238.195  │ -uacomment=w1                │ None     │          │
│ 2     │ bitcoindevproject/bitcoin:26.0 │ 100.156.168.254 │ -uacomment=w2 -debug=mempool │ None     │          │
│ 3     │ 27.0                           │ 100.241.115.233 │ -uacomment=w3                │ None     │          │
│ 4     │ 27.0                           │ 100.27.63.92    │ -uacomment=w4                │ None     │          │
│ 5     │ 27.0                           │ 100.35.255.94   │ -uacomment=w5                │ None     │          │
│ 6     │ 27.0                           │ 100.183.131.60  │ -uacomment=w6                │ None     │          │
│ 7     │ 27.0                           │ 100.3.36.90     │ -uacomment=w7                │ None     │          │
│ 8     │ 27.0                           │ 100.222.119.143 │ -uacomment=w8                │ None     │          │
│ 9     │ 27.0                           │ 100.182.187.176 │ -uacomment=w9                │ None     │          │
│ 10    │ 27.0                           │ 100.137.132.18  │ -uacomment=w10               │ None     │          │
│ 11    │ 27.0                           │ 100.23.241.102  │ -uacomment=w11               │ None     │          │
└───────┴────────────────────────────────┴─────────────────┴──────────────────────────────┴──────────┴──────────┘
(.venv) grant@remove-compose:~/warnet$ kubectl get all
NAME                     READY   STATUS      RESTARTS       AGE
pod/rpc-0                1/1     Running     3 (3m5s ago)   86m
pod/warnet-tank-000000   0/1     Completed   0              86m
pod/warnet-tank-000001   0/1     Completed   0              86m
pod/warnet-tank-000002   0/1     Completed   0              86m
pod/warnet-tank-000003   0/1     Completed   0              86m
pod/warnet-tank-000004   0/1     Completed   1              86m
pod/warnet-tank-000005   0/1     Completed   1              86m
pod/warnet-tank-000006   0/1     Completed   1              86m
pod/warnet-tank-000007   0/1     Completed   0              86m
pod/warnet-tank-000008   0/1     Completed   0              86m
pod/warnet-tank-000009   0/1     Completed   0              86m
pod/warnet-tank-000010   0/1     Completed   0              86m
pod/warnet-tank-000011   0/1     Completed   0              86m

NAME                                 TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                            AGE
service/rpc                          ClusterIP   None             <none>        9276/TCP                                           86m
service/warnet-tank-000000-service   ClusterIP   10.97.130.76     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m
service/warnet-tank-000001-service   ClusterIP   10.98.227.82     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m
service/warnet-tank-000002-service   ClusterIP   10.104.158.18    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m
service/warnet-tank-000003-service   ClusterIP   10.102.166.85    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m
service/warnet-tank-000004-service   ClusterIP   10.103.177.164   <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m
service/warnet-tank-000005-service   ClusterIP   10.101.119.229   <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m
service/warnet-tank-000006-service   ClusterIP   10.109.216.81    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m
service/warnet-tank-000007-service   ClusterIP   10.100.56.101    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m
service/warnet-tank-000008-service   ClusterIP   10.101.155.49    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m
service/warnet-tank-000009-service   ClusterIP   10.106.24.140    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m
service/warnet-tank-000010-service   ClusterIP   10.96.21.131     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m
service/warnet-tank-000011-service   ClusterIP   10.99.47.99      <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   86m

NAME                   READY   AGE
statefulset.apps/rpc   1/1     86m

````

</details>

Reply:
Interesting, thanks. How can I re-create this state? What kind of issues can `kubectl` have when it returns `Running` to then get to here?

To me, the error posted above looks like a general connectivity issue (which we should still handle, but is not being changed/addressed in this PR)?

Reply:
To recreate the state, I spin up a fresh gcloud vm, install git, vim, python3, and python3-venv. Clone the repo. Then I run `quick_start.sh`. Install what it asks. Then I do a `just installlogging`. 

I attempted to run through the steps over the last hour or so, but ...
<details>
<summary>... I'm running into timeouts</summary>

```shell
+ minikube start --mount --mount-string=/home/grantpochmayer/warnet:/mnt/src
😄  minikube v1.33.1 on Debian 12.5 (amd64)
✨  Using the docker driver based on existing profile
👍  Starting "minikube" primary control-plane node in "minikube" cluster
🚜  Pulling base image v0.0.44 ...
🔄  Restarting existing docker container for "minikube" ...
🐳  Preparing Kubernetes v1.30.0 on Docker 26.1.1 ...
🔎  Verifying Kubernetes components...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🌟  Enabled addons: default-storageclass, storage-provisioner
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
+ docker build -t warnet/dev -f src/templates/rpc/Dockerfile_rpc_dev . --load
[+] Building 392.2s (6/10)                                                                                                             docker:default
 => [internal] load build definition from Dockerfile_rpc_dev                                                                                     0.0s
 => => transferring dockerfile: 1.04kB                                                                                                           0.0s
 => [internal] load metadata for docker.io/library/python:3.12-slim                                                                              0.3s
 => [internal] load .dockerignore                                                                                                                0.0s
 => => transferring context: 173B                                                                                                                0.0s
 => CACHED [1/6] FROM docker.io/library/python:3.12-slim@sha256:2fba8e70a87bcc9f6edd20dda0a1d4adb32046d2acbca7361bc61da5a106a914                 0.0s
 => [internal] load build context                                                                                                                0.0s
 => => transferring context: 146B                                                                                                                0.0s
 => ERROR [2/6] RUN apt-get update &&     apt-get install -y procps openssh-client &&     rm -rf /var/lib/apt/lists/*                          391.9s
------                                                                                                                                                
 > [2/6] RUN apt-get update &&     apt-get install -y procps openssh-client &&     rm -rf /var/lib/apt/lists/*:                                       
0.431 Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]                                                                                  
30.46 Ign:1 http://deb.debian.org/debian bookworm InRelease                                                                                           
30.54 Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]                                                                         
60.57 Ign:2 http://deb.debian.org/debian bookworm-updates InRelease                                                                                   
90.75 Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
120.8 Ign:3 http://deb.debian.org/debian-security bookworm-security InRelease
120.9 Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
150.9 Ign:1 http://deb.debian.org/debian bookworm InRelease
151.0 Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
181.0 Ign:2 http://deb.debian.org/debian bookworm-updates InRelease
181.1 Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
211.1 Ign:3 http://deb.debian.org/debian-security bookworm-security InRelease
211.2 Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
241.3 Ign:1 http://deb.debian.org/debian bookworm InRelease
241.3 Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
271.4 Ign:2 http://deb.debian.org/debian bookworm-updates InRelease
271.5 Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
301.5 Ign:3 http://deb.debian.org/debian-security bookworm-security InRelease
301.6 Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
331.6 Err:1 http://deb.debian.org/debian bookworm InRelease
331.6   Connection timed out [IP: 199.232.98.132 80]
331.7 Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
361.7 Err:2 http://deb.debian.org/debian bookworm-updates InRelease
361.7   Connection timed out [IP: 199.232.98.132 80]
361.8 Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
391.8 Err:3 http://deb.debian.org/debian-security bookworm-security InRelease
391.8   Connection timed out [IP: 199.232.98.132 80]
391.8 Reading package lists...
391.8 W: Failed to fetch http://deb.debian.org/debian/dists/bookworm/InRelease  Connection timed out [IP: 199.232.98.132 80]
391.8 W: Failed to fetch http://deb.debian.org/debian/dists/bookworm-updates/InRelease  Connection timed out [IP: 199.232.98.132 80]
391.8 W: Failed to fetch http://deb.debian.org/debian-security/dists/bookworm-security/InRelease  Connection timed out [IP: 199.232.98.132 80]
391.8 W: Some index files failed to download. They have been ignored, or old ones used instead.
391.8 Reading package lists...
391.8 Building dependency tree...
391.8 Reading state information...
391.8 E: Unable to locate package procps
391.8 E: Unable to locate package openssh-client
------
Dockerfile_rpc_dev:5
--------------------
   4 |     # Install procps, which includes pgrep
   5 | >>> RUN apt-get update && \
   6 | >>>     apt-get install -y procps openssh-client && \
   7 | >>>     rm -rf /var/lib/apt/lists/*
   8 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get update &&     apt-get install -y procps openssh-client &&     rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100
error: Recipe `start` failed with exit code 1    
```

</details>

I can try again later assuming this is some kind of intermittent network issue on their end.

The first time I ran `just installlogging` on the remove-compose branch, `kubectl get all` failed with a timeout (or similar), yet `minikube status` came back with "Running" according to my script. Maybe that means there is a bug in minikube. Will look into this further.

Reply:
I see, that does seem to look more like a general network issue on the vm (seems unlikely) or debian package repo (or something like that) to me...

Let me know what you find out anyhow, and I can make any necessary changes :)

Reply:
I got it recreated in google cloud. The issue seems to be that once we try to do `just installlogging`, warnet fails but tries to keep going. However, despite this, `minikube status` still reports to us that it is `Running`.

<details>
<summary>The setup</summary>

``` shell
Linux instance-20240614-220500 6.1.0-21-cloud-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.90-1 (2024-05-03) x86_64

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.

grant@instance-20240614-220500:~$ sudo apt install vim git python3-venv python3
grant@instance-20240614-220500:~$ git clone https://github.com/willcl-ark/warnet.git
grant@instance-20240614-220500:~$ cd warnet/
grant@instance-20240614-220500:~/warnet$ ./quick_start.sh 

[Get everything installed. Do not forget ito install Helm]

grant@instance-20240614-220500:~/warnet$ ./quick_start.sh 

   ╭───────────────────────────────────╮
   │  Welcome to the Warnet Quickstart │
   ╰───────────────────────────────────╯

    Let's find out if your system has what it takes to run Warnet...

 ⭐️ Currently in the warnet directory
 ⭐️ Found minikube: /usr/local/bin/minikube 
 ⭐️ Found kubectl: /usr/local/bin/kubectl 
 ⭐️ Found docker: /usr/bin/docker
 ⭐️ Found grant in the docker group
 ⭐️ Found just: /home/grant/bin/just 
 ⭐️ Found python3: /usr/bin/python3 
 ⭐️ Found venv: a python3 module
 ⭐️ Found BPF: Berkeley Packet Filters appear enabled

    Let's try to spin up a python virtual environment...

    Creating a new virtual environment...

 ⭐️ venv: The python virtual environment looks good

    Let's install warnet into that virtual environment...

[snip]

 ⭐️ warnet: We installed Warnet in the virtual environment

    Now, let's get the Warnet started...

+ minikube delete
🙄  "minikube" profile does not exist, trying anyways.
💀  Removed all traces of the "minikube" cluster.
+ minikube start --mount --mount-string=/home/grant/warnet:/mnt/src
😄  minikube v1.33.1 on Debian 12.5 (amd64)
✨  Automatically selected the docker driver
📌  Using Docker driver with root privileges
👍  Starting "minikube" primary control-plane node in "minikube" cluster
🚜  Pulling base image v0.0.44 ...
💾  Downloading Kubernetes v1.30.0 preload ...
    > preloaded-images-k8s-v18-v1...:  342.90 MiB / 342.90 MiB  100.00% 224.91 
    > gcr.io/k8s-minikube/kicbase...:  481.58 MiB / 481.58 MiB  100.00% 94.34 M
🔥  Creating docker container (CPUs=2, Memory=2200MB) ...
🐳  Preparing Kubernetes v1.30.0 on Docker 26.1.1 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔗  Configuring bridge CNI (Container Networking Interface) ...
🔎  Verifying Kubernetes components...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
+ kubectl apply -f src/templates/rpc/namespace.yaml
namespace/warnet created
namespace/warnet-logging created
+ kubectl apply -f src/templates/rpc/rbac-config.yaml
role.rbac.authorization.k8s.io/pod-reader-exec created
rolebinding.rbac.authorization.k8s.io/read-exec-pods created
clusterrole.rbac.authorization.k8s.io/crd-viewer created
clusterrolebinding.rbac.authorization.k8s.io/crd-viewer-binding created
+ kubectl apply -f src/templates/rpc/warnet-rpc-service.yaml
service/rpc created
+ kubectl apply -f src/templates/rpc/warnet-rpc-statefulset-dev.yaml
statefulset.apps/rpc created
+ kubectl config set-context --current --namespace=warnet
Context "minikube" modified.
+ kubectl get pod rpc-0 --namespace=warnet
Error from server (NotFound): pods "rpc-0" not found
+ echo 'Waiting for server to find pod rpc-0...'
Waiting for server to find pod rpc-0...
+ sleep 4
+ kubectl get pod rpc-0 --namespace=warnet
Error from server (NotFound): pods "rpc-0" not found
+ echo 'Waiting for server to find pod rpc-0...'
Waiting for server to find pod rpc-0...
+ sleep 4
+ kubectl get pod rpc-0 --namespace=warnet
Error from server (NotFound): pods "rpc-0" not found
+ echo 'Waiting for server to find pod rpc-0...'
Waiting for server to find pod rpc-0...
+ sleep 4
+ kubectl get pod rpc-0 --namespace=warnet
NAME    READY   STATUS              RESTARTS   AGE
rpc-0   0/1     ContainerCreating   0          3s
+ echo '⏲️ This could take a minute or so.'
⏲️ This could take a minute or so.
+ kubectl wait --for=condition=Ready --timeout=2m pod rpc-0
pod/rpc-0 condition met
+ echo Done...
Done...
kubectl port-forward svc/rpc 9276:9276
Forwarding from 127.0.0.1:9276 -> 9276
Forwarding from [::1]:9276 -> 9276
Handling connection for 9276
Warnet:
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Temp dir                    ┃ Bitcoin network ┃ Docker network ┃ Subnet      ┃ Graph                                   ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ /root/.warnet/warnet/warnet │ regtest         │ warnet         │ 100.0.0.0/8 │ MultiDiGraph with 12 nodes and 12 edges │
└─────────────────────────────┴─────────────────┴────────────────┴─────────────┴─────────────────────────────────────────┘

Tanks:
┏━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━┓
┃ Index ┃ Version                        ┃ IPv4            ┃ bitcoin conf                 ┃ tc_netem ┃ LN Image ┃
┡━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━┩
│ 0     │ 26.0                           │ 100.27.105.109  │ -uacomment=w0                │ None     │          │
│ 1     │ 26.0                           │ 100.4.211.135   │ -uacomment=w1                │ None     │          │
│ 2     │ bitcoindevproject/bitcoin:26.0 │ 100.11.210.129  │ -uacomment=w2 -debug=mempool │ None     │          │
│ 3     │ 26.0                           │ 100.148.53.103  │ -uacomment=w3                │ None     │          │
│ 4     │ 26.0                           │ 100.129.58.21   │ -uacomment=w4                │ None     │          │
│ 5     │ 26.0                           │ 100.8.247.123   │ -uacomment=w5                │ None     │          │
│ 6     │ 26.0                           │ 100.19.141.56   │ -uacomment=w6                │ None     │          │
│ 7     │ 26.0                           │ 100.157.37.230  │ -uacomment=w7                │ None     │          │
│ 8     │ 26.0                           │ 100.58.217.65   │ -uacomment=w8                │ None     │          │
│ 9     │ 26.0                           │ 100.136.233.245 │ -uacomment=w9                │ None     │          │
│ 10    │ 26.0                           │ 100.212.213.181 │ -uacomment=w10               │ None     │          │
│ 11    │ 26.0                           │ 100.132.18.230  │ -uacomment=w11               │ None     │          │
└───────┴────────────────────────────────┴─────────────────┴──────────────────────────────┴──────────┴──────────┘
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
Handling connection for 9276
🥳
Run the following command to enter into the python virtual environment...
    source .venv/bin/activate
   After that, you can run warcli help to start running Warnet commands.
```

</details>

<details>
<summary>Try installlogging</summary>

```shell
grant@instance-20240614-220500:~/warnet$ source .venv/bin/activate

(.venv) grant@instance-20240614-220500:~/warnet$ warcli network status

Handling connection for 9276
Tank: 0         Bitcoin: running
Tank: 1         Bitcoin: running
Tank: 2         Bitcoin: running
Tank: 3         Bitcoin: running
Tank: 4         Bitcoin: running
Tank: 5         Bitcoin: running
Tank: 6         Bitcoin: running
Tank: 7         Bitcoin: running
Tank: 8         Bitcoin: running
Tank: 9         Bitcoin: running
Tank: 10        Bitcoin: running
Tank: 11        Bitcoin: running

(.venv) grant@instance-20240614-220500:~/warnet$ warcli network connected
Handling connection for 9276
True

(.venv) grant@instance-20240614-220500:~/warnet$ just installlogging
./src/templates/k8s/install_logging.sh
"grafana" has been added to your repositories
"prometheus-community" has been added to your repositories
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "grafana" chart repository
...Successfully got an update from the "prometheus-community" chart repository
Update Complete. ⎈Happy Helming!⎈
Release "loki" does not exist. Installing it now.
W0614 22:17:48.912495   19048 reflector.go:470] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: watch of *unstructured.Unstructured ended with: an error on the server ("unable to decode an event from the watch stream: http2: client connection lost") has prevented the request from succeeding
W0614 22:17:59.900652   19048 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
I0614 22:17:59.900911   19048 trace.go:236] Trace[825096021]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 22:17:49.893) (total time: 10007ms):
Trace[825096021]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout 10007ms (22:17:59.900)
Trace[825096021]: [10.00730579s] [10.00730579s] END
E0614 22:17:59.901152   19048 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
W0614 22:18:12.908742   19048 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
I0614 22:18:12.908840   19048 trace.go:236] Trace[471539342]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 22:18:02.899) (total time: 10009ms):
Trace[471539342]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout 10009ms (22:18:12.908)
Trace[471539342]: [10.009504706s] [10.009504706s] END
E0614 22:18:12.908868   19048 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
W0614 22:18:27.662844   19048 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
I0614 22:18:27.662935   19048 trace.go:236] Trace[1869785041]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 22:18:17.658) (total time: 10004ms):
Trace[1869785041]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout 10004ms (22:18:27.662)
Trace[1869785041]: [10.004495867s] [10.004495867s] END
E0614 22:18:27.662958   19048 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
W0614 22:18:50.185757   19048 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
I0614 22:18:50.185916   19048 trace.go:236] Trace[1033846673]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 22:18:40.180) (total time: 10005ms):
Trace[1033846673]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout 10005ms (22:18:50.185)
Trace[1033846673]: [10.005269892s] [10.005269892s] END
E0614 22:18:50.185945   19048 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
W0614 22:19:19.873610   19048 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
I0614 22:19:19.873935   19048 trace.go:236] Trace[516593810]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 22:19:09.871) (total time: 10001ms):
Trace[516593810]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout 10001ms (22:19:19.873)
Trace[516593810]: [10.001837876s] [10.001837876s] END
E0614 22:19:19.873976   19048 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
W0614 22:20:00.715906   19048 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
I0614 22:20:00.716300   19048 trace.go:236] Trace[1712799558]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 22:19:50.708) (total time: 10007ms):
Trace[1712799558]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout 10007ms (22:20:00.715)
Trace[1712799558]: [10.007781118s] [10.007781118s] END
E0614 22:20:00.716344   19048 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
W0614 22:21:01.905528   19048 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
I0614 22:21:01.905627   19048 trace.go:236] Trace[1951240226]: "Reflector ListAndWatch" name:k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232 (14-Jun-2024 22:20:51.896) (total time: 10008ms):
Trace[1951240226]: ---"Objects listed" error:Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout 10008ms (22:21:01.905)
Trace[1951240226]: [10.008638451s] [10.008638451s] END
E0614 22:21:01.905654   19048 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Get "https://192.168.49.2:8443/apis/batch/v1/namespaces/warnet-logging/jobs?fieldSelector=metadata.name%3Dloki-minio-make-user-job&resourceVersion=1393": net/http: TLS handshake timeout
Error: failed post-install: 1 error occurred:
        * timed out waiting for the condition


error: Recipe `installlogging` failed on line 87 with exit code 1

(.venv) grant@instance-20240614-220500:~/warnet$ kubectl get all
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout


(.venv) grant@instance-20240614-220500:~/warnet$ minikube status
minikube
type: Control Plane
host: Running
kubelet: Running
apiserver: Stopped
kubeconfig: Configured

(.venv) grant@instance-20240614-220500:~/warnet$ kubectl get all


NAME                     READY   STATUS    RESTARTS      AGE
pod/rpc-0                0/1     Running   1 (16m ago)   17m
pod/warnet-tank-000000   0/1     Running   1 (86s ago)   15m
pod/warnet-tank-000001   0/1     Running   1 (86s ago)   15m
pod/warnet-tank-000002   0/1     Running   1 (86s ago)   15m
pod/warnet-tank-000003   0/1     Running   1 (86s ago)   15m
pod/warnet-tank-000004   1/1     Running   1 (86s ago)   15m
pod/warnet-tank-000005   0/1     Running   1 (86s ago)   15m
pod/warnet-tank-000006   0/1     Running   1 (86s ago)   15m
pod/warnet-tank-000007   0/1     Running   1 (86s ago)   15m
pod/warnet-tank-000008   0/1     Running   1 (86s ago)   15m
pod/warnet-tank-000009   0/1     Running   1 (86s ago)   15m
pod/warnet-tank-000010   0/1     Running   1 (86s ago)   15m
pod/warnet-tank-000011   0/1     Running   1 (86s ago)   15m

NAME                                 TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                            AGE
service/rpc                          ClusterIP   None             <none>        9276/TCP                                           17m
service/warnet-tank-000000-service   ClusterIP   10.96.73.109     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m
service/warnet-tank-000001-service   ClusterIP   10.106.44.187    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m
service/warnet-tank-000002-service   ClusterIP   10.102.242.18    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m
service/warnet-tank-000003-service   ClusterIP   10.96.105.191    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m
service/warnet-tank-000004-service   ClusterIP   10.98.248.41     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m
service/warnet-tank-000005-service   ClusterIP   10.110.234.51    <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m
service/warnet-tank-000006-service   ClusterIP   10.103.204.232   <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m
service/warnet-tank-000007-service   ClusterIP   10.96.202.93     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m
service/warnet-tank-000008-service   ClusterIP   10.111.79.19     <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m
service/warnet-tank-000009-service   ClusterIP   10.110.246.113   <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m
service/warnet-tank-000010-service   ClusterIP   10.111.254.132   <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m
service/warnet-tank-000011-service   ClusterIP   10.103.148.185   <none>        18444/TCP,18443/TCP,28332/TCP,28333/TCP,9332/TCP   15m

NAME                   READY   AGE
statefulset.apps/rpc   0/1     17m

(.venv) grant@instance-20240614-220500:~/warnet$ kubectl get all
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout

(.venv) grant@instance-20240614-220500:~/warnet$ kubectl get all
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: http2: client connection lost
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout

(.venv) grant@instance-20240614-220500:~/warnet$ minikube status
minikube
type: Control Plane
host: Running
kubelet: Running
apiserver: Stopped
kubeconfig: Configured

(.venv) grant@instance-20240614-220500:~/warnet$ kubectl get all
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
Unable to connect to the server: net/http: TLS handshake timeout
```
</details>

Reply:
I agree we should detect this failure somehow, but the root cause does still seem to be connectivity issue to wherever we download Loki from... 🤔

Reply:
I'm fine with punting on this. I may want to do a `minikube delete` in  `quick_start.sh` via another PR in order to provide a way to nuke the current setup if it becomes unstable.